doc: Updated H20 GS and ABI comp (#19560)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.9.0-nrf54h20.rst

.. _migration_2.9.0-nrf54h20:

Migration guide for |NCS| v2.9.0-nRF54H20 (Working draft)
#########################################################

.. contents::
   :local:
   :depth: 3

This document describes the changes required or recommended when migrating your nRF54H20 application from the |NCS| v2.8.0 to the |NCS| v2.9.0-nRF54H20.

.. HOWTO

   Add changes in the following format:

   Component (for example, application, sample or libraries)
   *********************************************************

   .. toggle::

      * Change1 and description
      * Change2 and description

.. _migration_2.9.0-nrf54h20_required:

Required changes
****************

The following changes are mandatory to make your application work in the same way as in previous releases.

nRF54H20
========

This section describes the changes specific to the nRF54H20 SoC and DK support in the |NCS|.
For more information on changes related to samples and applications usage on the nRF54H20 DK, see :ref:`migration_2.9.0-nrf54h20_required_nrf54h`.

DK compatibility
----------------

.. toggle::

>    * The |NCS| v2.9.0-nRF54H20 is compatible only with the Engineering C - v0.9.0 and later revisions of the nRF54H20 DK, PCA10175.
>     Check the version number on your DK's sticker to verify its compatibility with the |NCS|.

Reviewer: ```suggestion
   * The |NCS| v2.9.0-nRF54H20 is compatible only with the Engineering C - v0.9.0 and later revisions of the nRF54H20 DK, PCA10175.
     Check the version number on your DK's sticker to verify its compatibility with the |NCS|.
```

Dependencies
------------

The following required dependencies for the nRF54H20 SoC and DK have been updated.

## >> ADD toolchain switch <<

nRF54H20 SoC binaries
+++++++++++++++++++++

.. toggle::

  * The *nRF54H20 SoC binaries* bundle has been updated to version 0.8.0.

    To update the SoC binaries bundle of your development kit while in Root of Trust, do the following:

    1. Download the `nRF54H20 SoC Binaries v0.8.0`_.

       .. note::
          On MacOS, ensure that the ZIP file is not unpacked automatically upon download.

    #. Purge the device as follows::

          nrfutil device recover --core Application --serial-number <serial_number>
          nrfutil device recover --core Network --serial-number <serial_number>

    #. Run ``west update``.

>     #. Move the correct :file:`.zip` bundle to a folder of your choice, then run nRF Util to program the binaries using one of the following commands:

Reviewer: ```suggestion
    #. Move the correct :file:`.zip` bundle to a folder of your choice, then run nRF Util to program the binaries using one of the following commands::
```

          nrfutil device x-suit-dfu --firmware nrf54h20_soc_binaries_v0.8.0.zip --serial-number <serial_number>

nrfutil
+++++++

.. toggle::

  * ``nrfutil`` has been updated to version 7.13.0.

    Install nRF Util version 7.13.0 as follows:

>       1. Download the nrfutil executable file from the `nRF Util development tool`_ product page.

Reviewer: ```suggestion
      1. Download the nRF Util executable file from the `nRF Util development tool`_ product page.
```

      #. Add nRF Util to the system path on Linux and MacOS, or environment variables on Windows, to run it from anywhere on the system.
         On Linux and MacOS, use one of the following options:

         * Add nRF Util's directory to the system path.
         * Move the file to a directory in the system path.

      #. On MacOS and Linux, give ``nrfutil`` execute permissions by typing ``chmod +x nrfutil`` in a terminal or using a file browser.
         This is typically a checkbox found under file properties.

>       #. On MacOS, to run the nrfutil executable you need to allow it in the system settings.

Reviewer: ```suggestion
      #. On MacOS, to run the nRF Util executable, you need to allow it in the system settings.
```

      #. Verify the version of the nRF Util installation on your machine by running the following command::

            nrfutil --version

      #. If your version is below 7.13.0, run the following command to update nRF Util::

            nrfutil self-upgrade

>          For more information, consult the `nRF Util`_ documentation.

Reviewer: ```suggestion
         For more information, see the `nRF Util`_ documentation.
```

    For more information, consult the `nRF Util`_ documentation.

nrfutil suit
++++++++++++

.. toggle::

  * ``nrfutil suit`` has been updated to version 0.9.0.

    Install the nRF Util ``suit`` command version 0.9.0 as follows::

       nrfutil install suit=0.9.0 --force

    For more information, consult the `nRF Util`_ documentation.

Samples and applications
========================

This section describes the changes related to samples and applications.

|no_changes_yet_note|

.. _migration_2.9.0-nrf54h20_required_nrf54h:

nRF54H20
--------

|no_changes_yet_note|

Libraries
=========

This section describes the changes related to libraries.

|no_changes_yet_note|

.. _migration_2.9.0-nrf54h20_recommended:

Recommended changes
*******************

The following changes are recommended for your application to work optimally after the migration.

Samples and applications
========================

This section describes the changes related to samples and applications.

Matter
------

This section describes the changes related to Matter support.

|no_changes_yet_note|

Libraries
=========

This section describes the changes related to libraries.

> |no_changes_yet_note|

Reviewer: Just a reminder to remove these entries, may be towards the end of this release? as this a nRF54H20 migration guide.
